code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -28,47 +28,43 @@
 	return ..()
 
 /mob/living/simple_animal/hostile/proc/FindTarget()
-
 	if(!faction) //No faction, no reason to attack anybody.
 		return null
 
 	var/atom/T = null
 	var/lowest_health = INFINITY // Max you can get
 	stop_automated_movement = 0
-
-	for(var/atom/A in targets)
-
-		if(A == src)
+	for(var/mob/living/L in targets)
+		if((L.faction == src.faction) && !attack_same)
+			continue
+		if(L in friends)
 			continue
 
-		var/atom/F = Found(A)
-		if(F)
-			T = F
-			break
+		if(!L.stat && (L.health < lowest_health))
+			lowest_health = L.health
+			T = L
+	if(isnull(T))
+		for(var/atom/A in targets)
 
-		if(isliving(A))
-			var/mob/living/L = A
-			if((L.faction == src.faction) && !attack_same)
+			if(A == src)
 				continue
-			if(L in friends)
-				continue
-
-			if(!L.stat && (L.health < lowest_health))
-				lowest_health = L.health
-				T = L
-				break
 
-		else if(istype(A, /obj/mecha)) // Our line of sight stuff was already done in ListTargets().
-			var/obj/mecha/M = A
-			if (M.occupant)
-				T = M
+			var/atom/F = Found(A)
+			if(F)
+				T = F
 				break
 
-		if(istype(A, /obj/machinery/bot))
-			var/obj/machinery/bot/B = A
-			if (B.health > 0)
-				T = B
-				break
+			else if(istype(A, /obj/mecha)) // Our line of sight stuff was already done in ListTargets().

```

Comment:
So, here's a thought. Generalize this. At the moment, you have relatively meaningless overrides which simply add the functionality in the form of:
````dm
if (istype(A, /some/path) && SomeValidity(A))
    T = A
    break
````

Which is a lot of pointless reinventing of the wheel. Instead, this could be generalized as:
````c
var/list/target_type_validator_map = list(/some/type = CALLBACK(src, /proc/someFunction), /other/path = TRUE) // this would be a class variable

// ...

for (var/atom/A in targets)
    if (target_type_validator_map[A.type])
        var/datum/callback/cb = target_type_validator_map[A.type]
        if (!istype(cb))
            T = A // if it's not a callback, then we don't have a validator, ergo, always prefer this target!
        else if (cb.Invoke(A))
            T = A // Validator returned TRUE, valid target, go nuts.
        break
````